gobblin-iceberg/src/main/java/org/apache/gobblin/iceberg/writer/IcebergMetadataWriter.java

```diff
@@ -671,14 +796,29 @@ public void flush(String dbName, String tableName) throws IOException {
       TableMetadata tableMetadata = tableMetadataMap.getOrDefault(tid, new TableMetadata());
       if (tableMetadata.transaction.isPresent()) {
         Transaction transaction = tableMetadata.transaction.get();
+        Map<String, String> props = tableMetadata.newProperties.or(
+            Maps.newHashMap(tableMetadata.lastProperties.or(getIcebergTable(tid).properties())));
         if (tableMetadata.appendFiles.isPresent()) {
           tableMetadata.appendFiles.get().commit();
+          if(tableMetadata.completenessEnabled) {
+            String topicName = props.get(TOPIC_NAME_KEY);
+            if(topicName == null) {
+              log.error(String.format("Not performing audit check. %s is null. Please set as table property of %s.%s",
+                  TOPIC_NAME_KEY, dbName, tableName));
+            } else {
+              long newCompletenessWatermark =
```

Comment:
Calling this on each flush may be excessive. If the table is complete upto the most recent hour (or whatever the window granularity is), then we should skip the check until the end of the hour.

Reply:
This cannot be done without using the current hour of the day.
So you are suggesting something like if(datepartition > previousWatermark && currentHour > datepartitionHour)
if prevWatermark = 9am
datepartition = 10am
currentHour = 10am (do not check audit system) vs 11 am (check audit system because GMIP may be lagging)

Reply:
My main idea is to optimize for the common case, which is that pipeline is caught up and data is current. If hour(currentTime) == prevWatermark: skip audit check, else (we are lagging) and hence, do the check.